server/pulp/server/controllers/repository.py

```diff
@@ -968,11 +968,67 @@ def publish(repo_id, dist_id, publish_config_override=None, scheduled_call_id=No
     # Fire events describing the publish state
     fire_manager = manager_factory.event_fire_manager()
     fire_manager.fire_repo_publish_started(repo_id, dist_id)
-    result = _do_publish(repo_obj, dist_id, dist_inst, transfer_repo, conduit, call_config)
+    result = check_publish(repo_obj, dist_id, dist_inst, transfer_repo, conduit, call_config)
     fire_manager.fire_repo_publish_finished(result)
     return result
 
 
+def check_publish(repo_obj, dist_id, dist_inst, transfer_repo, conduit, call_config):
+    """
+    Check if the publish should be a no operation and therefore skipped.
+
+    :param repo_obj: repository object
+    :type  repo_obj: pulp.server.db.model.Repository
+    :param dist_id: identifies the distributor
+    :type  dist_id: str
+    :param dist_inst: instance of the distributor
+    :type  dist_inst: dict
```

Comment:
it is correct, because `_get_distributor_instance_and_config()` returns distributor instance and its config.